lib/db_connection/connection.ex

```diff
@@ -169,23 +169,25 @@ defmodule DBConnection.Connection do
         mon = Process.monitor(pid)
         handle_checkout({ref, mon}, timeout, from, state, s)
       %{client: :closed} ->
-        err = DBConnection.Error.exception("connection not available")
+        message = "connection not available because of disconnection"
+        err = DBConnection.Error.exception(message)
         {:reply, {:error, err}, s}
       %{queue: queue} when queue? == true ->
         client = {ref, Process.monitor(pid)}
         queue = :queue.in({client, timeout, from}, queue)
         {:noreply, %{s | queue: queue}}
       _ when queue? == false ->
-        err = DBConnection.Error.exception("connection not available")
+        message = "connection not available because of queue"
```

Comment:
This can only occur when `queue: false` is used. Otherwise the connection will queue the request until the `:DOWN` arrives and then receive an disconnected error.

Poolboy is simpler than sbroker so I would probably recommend poolboy unless requiring the queuing features (max queue length, codel management or rescue-able timeout) provided by sbroker. I will try to add a comparison of the pools to the README today.